packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1946,6 +2026,13 @@ export class KeyringController extends BaseController<
     for (const serializedKeyring of serializedKeyrings) {
       await this.#restoreKeyring(serializedKeyring);
     }
+
+    if (this.state.keyringsMetadata.length > this.#keyrings.length) {
+      this.update((state) => {
+        // remove metadata from the end of the array to have the same length as the keyrings array
+        state.keyringsMetadata = state.keyringsMetadata.slice(0, -1 * (state.keyringsMetadata.length - this.#keyrings.length));
+      })
+    }
```

Comment:
Hmm, this feels like an unexpected situation where I'm not sure it would be safe to 'just delete the last ones'. Maybe we should just throw an error here?